vehicle_gateway_python_helpers/vehicle_gateway_python_helpers/helpers.py

```diff
@@ -0,0 +1,177 @@
+# limitations under the License.
+
+from distutils.dir_util import copy_tree
+
+from ament_index_python.packages import get_package_share_directory
+from launch import LaunchContext, Substitution, SomeSubstitutionsType
+from launch.actions import ExecuteProcess
+
+import tempfile
+import os
+import xml.etree.ElementTree as ET
+from typing import List
+
+
+def get_px4_dir():
+    return get_package_share_directory('px4_sim')
+
+
+def seed_rootfs(rootfs):
+    px4_dir = get_px4_dir()
+    print(f'seeding rootfs at {rootfs} from {px4_dir}')
+    copy_tree(px4_dir, rootfs)
+
+
+def get_px4_process(drone_id, additional_env):
+    rootfs = tempfile.TemporaryDirectory()
+    rc_script = os.path.join(get_px4_dir(), 'etc/init.d-posix/rcS')
+    print('using rootfs ', rootfs.name)
+    seed_rootfs(rootfs.name)
+
+    run_px4 = ExecuteProcess(
+        cmd=['px4', '%s/ROMFS/px4fmu_common' % rootfs.name,
+             '-s', rc_script,
+             '-i', drone_id,
+             '-d'],
+        cwd=get_px4_dir(),
+        additional_env=additional_env,
+        output='screen')
+
+    return run_px4
+
+
+class WorldPoseFromSdfFrame(Substitution):
+    """Substitution that retrieves a frame from SDF."""
+
+    def __init__(self,
+                 frame_name: SomeSubstitutionsType,
+                 world_name: SomeSubstitutionsType,
+                 model_pose: SomeSubstitutionsType,
+                 coor_name: str) -> None:
+        super().__init__()
+
+        from launch.utilities import normalize_to_list_of_substitutions  # import here to avoid loop
+        self.__frame_name = normalize_to_list_of_substitutions(frame_name)
+        self.__world_name = normalize_to_list_of_substitutions(world_name)
+        self.__model_pose = normalize_to_list_of_substitutions(model_pose)
+        self.__x = '0.0'
+        self.__y = '0.0'
+        self.__z = '0.3'
+        self.__roll = '0.0'
+        self.__pitch = '0.0'
+        self.__yaw = '0.0'
+        self.__coord_name = coor_name
+
+    @property
+    def model_pose(self) -> List[Substitution]:
+        """Getter for model pose."""
+        return self.__model_pose
+
+    @property
+    def frame_name(self) -> List[Substitution]:
+        """Getter for frame name."""
+        return self.__frame_name
+
+    @property
+    def world_name(self) -> List[Substitution]:
+        """Getter for world name."""
+        return self.__world_name
+
+    def parseCoords(self, strCoords: str, key: str, strSplit: str):
+        x, y, z, roll, pitch, yaw = strCoords.split(strSplit)
+        if (self.__coord_name == 'x'):
+            return str(x)
+        if (self.__coord_name == 'y'):
+            return str(y)
+        if (self.__coord_name == 'z'):
+            return str(z)
+        if (self.__coord_name == 'roll'):
+            return str(roll)
+        if (self.__coord_name == 'pitch'):
+            return str(pitch)
+        if (self.__coord_name == 'yaw'):
+            return str(yaw)
```

Comment:
It doesn't seem like `key` is used here. I assume you mean to use `key` in place of `self.__coord_name`.

Also, I think it'd be good to use a `Literal` for the key. This will give you some nice autocomplete and can make it easier to handle the cases. For example:
```
key: Literal["x", "y", "z", "roll", "pitch", "yaw"]
```
https://docs.python.org/3/library/typing.html#typing.Literal